packages/seacas/libraries/ioss/src/Ioss_ChangeSet.h

```diff
@@ -0,0 +1,101 @@
+// Copyright(C) 2024 National Technology & Engineering Solutions
+// of Sandia, LLC (NTESS).  Under the terms of Contract DE-NA0003525 with
+// NTESS, the U.S. Government retains certain rights in this software.
+//
+// See packages/seacas/LICENSE for details
+
+#pragma once
+
+#include "Ioss_DatabaseIO.h"
+#include "Ioss_DBUsage.h"
+#include "Ioss_Region.h"
+#include "Ioss_ParallelUtils.h"
+
+#include <utility>
+#include <climits>
+#include <cstddef>
+#include <iostream>
+#include <sstream>
+#include <iomanip>
+#include <string>
+#include <functional>
+#include <map>
+
+namespace Ioss {
+
+using FileNameGenerator = std::function<std::string(const std::string& baseFileName, unsigned step)>;
+
+enum ChangeSetFormat {
+  CHANGE_SET_NONE               = (      0),
+  CHANGE_SET_INTERNAL_FILES     = (1U << 0),
+  CHANGE_SET_LINEAR_MULTI_FILES = (1U << 1),
+  CHANGE_SET_CYCLIC_MULTI_FILES = (1U << 2)
+};
+
+std::pair<std::string, Ioss::DatabaseIO*>
+expand_topology_files(FileNameGenerator generator,
+                      const Ioss::ParallelUtils &util,
+                      const std::string& basename, const std::string& db_type,
+                      const Ioss::PropertyManager& properties,
+                      Ioss::DatabaseUsage usage, int step);
+
+std::string
+expand_topology_files(FileNameGenerator generator,
+                      const Ioss::ParallelUtils &util,
+                      const std::string& basename,
+                      const Ioss::PropertyManager& properties,
+                      int step);
+
+FileNameGenerator construct_cyclic_filename_generator(unsigned cyclicCount);
+FileNameGenerator construct_linear_filename_generator();
+
+class IOSS_EXPORT ChangeSet {
+public:
+  ChangeSet(Ioss::Region *region);
+  ChangeSet(Ioss::DatabaseIO* db, const std::string& dbName, const std::string& dbType, unsigned fileCyclicCount);
+
+  virtual ~ChangeSet();
+
+  IOSS_NODISCARD unsigned supported_formats() const { return m_supportedFormats; }
+  IOSS_NODISCARD unsigned database_format() const { return m_databaseFormat; }
+
+  virtual void populate_change_sets();
+
+  IOSS_NODISCARD virtual DatabaseIO* open_change_set(unsigned index, Ioss::DatabaseUsage usage);
+  virtual void close_change_set(unsigned index);
+
+  IOSS_NODISCARD unsigned size() const { return m_changeSetNames.size(); }
+  IOSS_NODISCARD const std::vector<std::string>& names() const { return m_changeSetNames; }
+  IOSS_NODISCARD std::string get_change_set_name(unsigned index) const;
+
+  IOSS_NODISCARD unsigned get_file_cyclic_count() const { return m_fileCyclicCount; }
+
+private:
+  ChangeSet() = delete;
```

Comment:
If deleted, no reason to put down in private.  Put up by other constructors so easier to see what exists and what doesn't.

Reply:
No problem